test/fail_compilation/goto2.d

```diff
@@ -48,3 +48,63 @@ void test1()
             break;
     }
 }
+
+/**************************************************
+https://issues.dlang.org/show_bug.cgi?id=11540
+goto label + try-catch-finally / with statement
+
+TEST_OUTPUT:
+---
+fail_compilation/goto2.d(1121): Error: cannot `goto` into `try` block
+fail_compilation/goto2.d(1143): Error: cannot `goto` into `try` block
```

Comment:
Why is this an error? It's not like you are entering an EH region.

Reply:
It's not mentioned in the spec but dmd explicitly disallows jumps into `try` blocks:

https://github.com/dlang/dmd/blob/0517fc93ceb9e74e979bbbc9de64eb64d610094c/src/dmd/statement.d#L2320

(This code wasn't skipped because `tryBody` was missing from the `Scope` used during `statementsem`.

Reply:
Actually, there is an [open bug report](https://issues.dlang.org/show_bug.cgi?id=13815) for this already.

Reply:
GDC (and LDC?) could disable probably disable [this code](https://github.com/dlang/dmd/blob/0517fc93ceb9e74e979bbbc9de64eb64d610094c/src/dmd/statement.d#L2315-L2329) if this problem is limited to DMD's backend.